### DISCUSSION TITLE
Questions about the logging format!

### DISCUSSION THREAD

**Discussion by ThereforeGames** [Mikubill/sd-webui-controlnet#1840] [Q&A]
Hello! A quick question for the CN maintainers.

I recently implemented the logging module in my extension, [Unprompted](https://github.com/ThereforeGames/unprompted). I decided to adopt the colors and string format of the ControlNet logger since it is probably the most popular extension to utilize the same module - I'm trying to ensure that the console output has a consistent look since AFAIK the WebUI itself does not prescribe a standard for this.

My question is, how were the colors and string format selected for ControlNet messages? They definitely look nice and include all the right information, but I'm wondering if there is a particular reason that "INFO" messages are green, or why hyphens are used as separators.

Just curious! Thanks.

**Comment by huchenlei** [comment]
You can reference the logger implementation here:
https://github.com/Mikubill/sd-webui-controlnet/blob/main/scripts/logging.py

I think when I implemented the logger, it was lllyasviel's idea that we should have some coloriazation on the log messages, so I implemented it. At least until now nobody is complaining about it.

**Reply by ThereforeGames** [reply]
Thanks - I'm not complaining, I quite like the look and have already yoinked these values for Unprompted, hah. I was just wondering if the colors were improvised or if there is some consensus in the greater Python community on log formatting.

I guess the adjacent question is, are there any plans to change the look in the future? I want to make sure my code is ready to adapt to such changes as clashing console messages is a pet peeve of mine.

**Reply by huchenlei** [reply]
I don't see a change in the future unless A1111 enforces a log format for all extensions.
